### PR TITLE
Added claimBounty command 

### LINKED ISSUE #377: Create command to claim bounty

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

Added claimBounty command to claim bounty for bounty hunter

Fixes #377

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cmd/claimBounty.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+		log.Error(err)
+		return core.NilHash, err
+	}
+
+	waitFor := big.NewInt(1).Sub(bountyLock.RedeemAfter, big.NewInt(int64(epoch)))
+	if waitFor.Cmp(big.NewInt(0)) == 1 {
+		log.Debug("Waiting for lock period to get over....")
+
+		//waiting till epoch reaches redeemAfter
+		utilsStruct.razorUtils.Sleep(time.Duration(waitFor.Int64()))
```
Comment (ashish10677) [inline cmd/claimBounty.go:100]: If wait for is 20, so this sleep is for 20 seconds, am I right?

File: cmd/claimBounty.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	rootCmd.AddCommand(claimBountyCmd)
+	var (
+		Address  string
+		Password string
+		BountyId uint32
+	)
+
+	claimBountyCmd.Flags().StringVarP(&Address, "address", "a", "", "address of the staker")
+	claimBountyCmd.Flags().StringVarP(&Password, "password", "", "", "password path of staker to protect the keystore")
+	claimBountyCmd.Flags().Uint32VarP(&BountyId, "bountyId", "", 0, "bountyId of the bounty hunter")
```
Comment (ashish10677) [inline cmd/claimBounty.go:130]: Can we put b for bountyId? Are we using `b` anywhere else?

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: cmd/claimBounty.go (hunk 1 above)
Comment (Yashk767) [reply cmd/claimBounty.go:100]: Ohh yes it is, I need it to multiply it by epoch length , and correct it to Sleep(time.Duration(waitFor.Int64()* epochLength)) correct ?

File: cmd/claimBounty.go (hunk 2 above)
Comment (Yashk767) [reply cmd/claimBounty.go:130]: Yes we are using it for bufferPercent.

File: cmd/claimBounty.go (hunk 1 above)
Comment (ashish10677) [reply cmd/claimBounty.go:100]: Yeah, but we also need block time. Can we write a function to calculate the time taken to mine a block by calculating the difference of block time between 2 latest blocks?
Then sleep duration will be `waitFor*epochLength*blockTime`.

We can get the latest block's time from:
https://github.com/ethereum/go-ethereum/blob/163f1665dd5074f64fd4b54360306837c6dee43d/core/types/block.go#L81

File: cmd/claimBounty.go (hunk 2 above)
Comment (ashish10677) [reply cmd/claimBounty.go:130]: Okay cool. No need then.

File: cmd/claimBounty.go (hunk 1 above)
Comment (Yashk767) [reply cmd/claimBounty.go:100]: Yes we can do that

File: cmd/claimBounty.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	}
+
+	if bountyLock.Amount.Cmp(big.NewInt(0)) == 0 {
+		err = errors.New("bounty amount is 0")
+		log.Error(err)
+		return core.NilHash, err
+	}
+
+	waitFor := big.NewInt(1).Sub(bountyLock.RedeemAfter, big.NewInt(int64(epoch)))
+	if waitFor.Cmp(big.NewInt(0)) == 1 {
```
Comment (ashish10677) [inline cmd/claimBounty.go:96]: I didn't understand this if statement. You are checking if the value of `waitFor == 0` then you are waiting???

File: cmd/claimBounty.go (hunk 1 above)
Comment (ashish10677) [reply cmd/claimBounty.go:100]: Also, even after doing that, we might not have reached the exact `RedeemAfter` period. Put a retry as well.

File: cmd/claimBounty.go (hunk 3 above)
Comment (Yashk767) [reply cmd/claimBounty.go:96]: waitFor is difference in redeemAfter epoch and current epoch.
If waitFor  > 0 than Cmp statement would return 1. So if waitFor is > 0 that means we need to waitFor some time, therefore we are waiting for that specific amount of epochs.

Comment (Yashk767) [reply cmd/claimBounty.go:96]: If waitFor == 0 , than we don't need to wait so we are directly calling redeemBlock function without waiting as we would already reach the redeemAfter epoch.

Comment (ashish10677) [reply cmd/claimBounty.go:96]: Okay got it.

File: cmd/claimBounty.go (hunk 1 above)
Comment (ashish10677) [reply cmd/claimBounty.go:100]: 2 retry for 1 epoch each.

Comment (Yashk767) [reply cmd/claimBounty.go:100]: okay

**Comment by ashish10677** [comment]
@Yashk767 can you update README as well?

**Review by ashish10677 — APPROVED** [review]
LGTM

**Pull request merged**